### PR TITLE
Adding junit output support for test-content command

### PR DESCRIPTION
**Pull request by kobymeir**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-8574

## Description
Adding junit output support for test-content command

Content: https://github.com/demisto/content/pull/29906

### CODE REVIEW

File: demisto_sdk/commands/test_content/TestContentClasses.py [hunk 1]
```diff
@@ -1068,18 +1142,6 @@ def print_test_summary(
         logging_module.info(
             f"Number of playbooks tested - {succeed_count + failed_count}"
         )
-        if failed_count:
```
Comment (kobymeir) [inline demisto_sdk/commands/test_content/TestContentClasses.py:1071]: moved down so the errors will be printed last

File: demisto_sdk/commands/test_content/TestContentClasses.py [hunk 2]
```diff
@@ -2782,135 +2998,21 @@ def create_manual_integration(self):
...
                 )
 
         except ApiException:
             self.build_context.logging_module.exception(
                 'No "Manual" integration found in XSIAM instance. '
                 "Please add it in order to create Manual Correlation Rule."
             )
-
-
-def replace_external_playbook_configuration(
```
Comment (kobymeir) [inline demisto_sdk/commands/test_content/TestContentClasses.py:2803]: the below functions were moved to be class functions

**Review by tkatzir — APPROVED** [review]
Very good!
Approved, with comments.

File: demisto_sdk/__main__.py [hunk 3]
```diff
@@ -2843,6 +2843,14 @@ def openapi_codegen(ctx, **kwargs):
     hidden=True,
 )
 @click.help_option("-h", "--help")
+@click.option(
+    "-a",
+    "--artifacts_path",
```
Comment (kobymeir) [inline demisto_sdk/__main__.py:2848]:
```suggestion
    "--artifacts-path",
```

File: demisto_sdk/commands/test_content/TestContentClasses.py [hunk 4]
```diff
@@ -584,59 +680,57 @@ def close_incident(self, client: DefaultApi, incident_id: str) -> bool:
...
                 "Failed to close incident, error trying to communicate with demisto server."
             )
             return False
 
-        if int(res[1]) != 200:
-            self.build_context.logging_module.warning(
-                f"Close incident failed with Status code {res[1]}"
+        if status_code != requests.codes.ok:
+            self.log_warning(
+                f"Close incident failed - response:{pformat(response)}, status code:{status_code} headers:{headers}"
```
[the remark this answers is not in the document]
Comment (kobymeir) [reply demisto_sdk/commands/test_content/TestContentClasses.py:695]: nope. tested

File: demisto_sdk/commands/test_content/TestContentClasses.py [hunk 5]
```diff
@@ -584,59 +680,57 @@ def close_incident(self, client: DefaultApi, incident_id: str) -> bool:
...
                 path=f"/investigation/{incident_id}/context",
                 body=body,
+                response_type="object",
             )
-            if int(res[1]) != 200:
-                self.build_context.logging_module.error(
-                    f"incident context fetch failed with Status code {res[1]}"
+            if status_code != requests.codes.ok:
+                self.log_error(
+                    f"incident context fetch failed - response:{pformat(response)}, status code:{status_code} headers:{headers}"
```
[the remark this answers is not in the document]
Comment (kobymeir) [reply demisto_sdk/commands/test_content/TestContentClasses.py:713]: nope, tested

File: demisto_sdk/commands/test_content/TestContentClasses.py [hunk 6]
```diff
@@ -696,23 +792,24 @@ def _get_all_integration_config(self, instances_ips: list) -> Optional[list]:
...
+    def get_all_installed_integrations_configurations(
+        self, server_url: str, timeout: int = 180, sleep_interval: int = 5
+    ) -> list:
         """
         Gets all integration configuration as it exists on the demisto server
         Args:
-            server_url: The url of the server to create integration in
+            sleep_interval: The interval to sleep between each poll.
+            timeout: The timeout to wait for the incident to be created.
+            server_url: The url of the server to create integration in.
```
Comment (kobymeir) [inline demisto_sdk/commands/test_content/TestContentClasses.py:806]:
```suggestion
            server_url: The url of the server to create integration in.
            timeout: The timeout to wait for the incident to be created.
            sleep_interval: The interval to sleep between each poll.
```

File: demisto_sdk/commands/test_content/TestContentClasses.py [hunk 7]
```diff
@@ -721,33 +818,30 @@ def get_all_installed_integrations_configurations(self, server_url: str) -> list
...
+                    body={},
+                    response_type="object",
+                )
+                if "configurations" in response:
+                    return response["configurations"]
+            except ApiException:
+                self.logging_module.exception(
+                    "failed to get all integrations configuration"
+                )
+            if time.time() > end_time:
```
Comment (kobymeir) [inline demisto_sdk/commands/test_content/TestContentClasses.py:838]:
```suggestion
            if time.time() > end_time:  # FIXME! log response
```

File: demisto_sdk/commands/test_content/TestContentClasses.py [hunk 8]
```diff
@@ -1027,26 +1122,34 @@ def add_proxy_related_test_data(self, proxy):
             self.empty_files.append(playbook_id)
 
     def create_result_files(self):
-        with open("./Tests/succeeded_tests.txt", "w") as succeeded_tests_file:
+        with open(
```
Comment (kobymeir) [inline demisto_sdk/commands/test_content/TestContentClasses.py:1125]:
```suggestion
        with open(  # FIXME! extract method: write_artifact_file
```

File: demisto_sdk/commands/test_content/TestContentClasses.py [hunk 9]
```diff
@@ -2405,50 +2620,46 @@ def _execute_mockable_test(self, proxy: MITMProxy):
...
-                with run_with_mock(
-                    proxy, self.playbook.configuration.playbook_id, record=True
-                ) as result_holder:
-                    status = self._incident_and_docker_test()
-                    self.playbook.configuration.number_of_executions += 1
-                    status = self._update_playbook_status(status, is_record_run=True)
-                    result_holder[RESULT] = status == PB_Status.SECOND_PLAYBACK_REQUIRED
-            else:
-                # If the integrations were not locked - the test has not finished it's execution
+            if not lock:
```
[the remark this answers is not in the document]
Comment (kobymeir) [reply demisto_sdk/commands/test_content/TestContentClasses.py:2649]: that's the implementation of acquire_test_lock. sadly not the best implementation...

File: demisto_sdk/commands/test_content/TestContentClasses.py [hunk 10]
```diff
@@ -2782,135 +2998,21 @@ def create_manual_integration(self):
...
-            res_raw_integration = demisto_client.generic_request_func(
+            response, status_code, headers = demisto_client.generic_request_func(
                 self=self.client,
                 method="PUT",
                 path="/settings/integration-conf",
                 body=manual_integration,
+                response_type="object",
             )
-            if int(res_raw_integration[1]) != 200:
+            if status_code != requests.codes.ok:
```
Comment (kobymeir) [inline demisto_sdk/commands/test_content/TestContentClasses.py:3008]:
```suggestion
            if status_code != requests.codes.ok:  # FIXME check 204
```

File: demisto_sdk/commands/test_content/TestContentClasses.py [hunk 11]
```diff
@@ -2782,135 +2998,21 @@ def create_manual_integration(self):
...
-            res_raw_integration = demisto_client.generic_request_func(
+            response, status_code, headers = demisto_client.generic_request_func(
                 self=self.client,
                 method="PUT",
                 path="/settings/integration-conf",
                 body=manual_integration,
+                response_type="object",
             )
-            if int(res_raw_integration[1]) != 200:
+            if status_code != requests.codes.ok:  # FIXME check 204
```
Comment (kobymeir) [inline demisto_sdk/commands/test_content/TestContentClasses.py:3008]:
```suggestion
            if status_code != requests.codes.ok:  # FIXME! check 204
```

**Pull request merged**